obsreport/obsreport_receiver.go

```diff
@@ -183,16 +217,44 @@ func ReceiverContext(
 	return ctx
 }
 
-// traceReceiveTraceDataOp creates the span used to trace the operation. Returning
-// the updated context and the created span.
-func traceReceiveTraceDataOp(
+// traceReceiveOp creates the span used to trace the operation. Returning
+// the updated context with the created span.
+func traceReceiveOp(
 	receiverCtx context.Context,
 	receiverName string,
 	transport string,
 	operationSuffix string,
+	opt ...StartReceiveOption,
 ) context.Context {
+	var opts StartReceiveOptions
+	for _, o := range opt {
+		o(&opts)
+	}
+
+	var sampleOption []trace.StartOption
+	if opts.AlwaysSample {
+		sampleOption = []trace.StartOption{
+			trace.WithSampler(trace.AlwaysSample()),
+		}
+	}
+
+	var ctx context.Context
+	var span *trace.Span
 	spanName := receiverPrefix + receiverName + operationSuffix
-	ctx, span := trace.StartSpan(receiverCtx, spanName)
+	if opts.LongLivedCtx == nil {
+		ctx, span = trace.StartSpan(receiverCtx, spanName, sampleOption...)
+	} else {
+		// Do not use LongLivedRPCCtx to start the span so this trace ends when
+		// the EndTraceDataReceiveOp is called. Here is safe to ignore the returned
+		// context since it is not used below.
+		_, span = trace.StartSpan(receiverCtx, spanName, sampleOption...)
```

Comment:
@bogdandrutu per previous code on OC receiver this context here can be ignored, please, double-check this assumption.

Reply:
Once you fix my previous comment about long lived being a bool this logic will change significantly and will be more clear.

Reply:
One small trick, when you know that the receiverCtx is long lived you set the option to become root for the start of the Span to start a new trace.

Reply:
I'm afraid I don't understand the trick: I only see the options `trace.WithSpanKind` and `trace.WithSampler`. I will do the other changes but this part will stay basically as it is until I understand the trick.

Reply:
What you have is ok, I thought we have the root option but could not find it.